Issue 4822: Updated to MSVC 2019

JCash opened the pull request:
Main reason was to update to new package format

JCash committed:
Issue 4822: Updated to MSVC 2019

Main reason was to update to new package format

JCash commented on build_tools/waf_dynamo.py:
```diff
@@ -1284,120 +1286,49 @@ def create_clang_wrapper(conf, exe):
     os.chmod(clang_wrapper_path, stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
     return clang_wrapper_path
 
-# Capture vcvarsall.bat environment variables for MSVC binaries, INCLUDE and LIB dir
```
Removed old unused stuff

JCash commented on build_tools/waf_dynamo.py:
```diff
@@ -1284,120 +1286,49 @@ def create_clang_wrapper(conf, exe):
 
-    includes = [os.path.join(msvcdir,'VC','include'),
-                os.path.join(msvcdir,'VC','atlmfc','include'),
+    msvc_version = [x for x in os.listdir(os.path.join(msvcdir,'VC','Tools','MSVC'))]
```
Finds the version automatically (assuming they stop moving folders around)

JCash commented on scripts/build.py:
```diff
@@ -65,9 +65,11 @@
-PACKAGES_WIN32_TOOLCHAIN="Microsoft-Visual-Studio-14-0"
+WINDOWS_SDK_10_VERSION="10.0.18362.0"
+WINDOWS_MSVC_2019_VERSION="14.25.28610"
+PACKAGES_WIN32_TOOLCHAIN="Microsoft-Visual-Studio-2019-{0}".format(WINDOWS_MSVC_2019_VERSION)
```
Now uses the same packages as the extender server.

JCash commented on scripts/build.py:
```diff
@@ -503,7 +505,7 @@ def download_sdk(url, targetfolder, strip_components=1):
             download_sdk( '%s/%s.tar.gz' % (self.package_path, PACKAGES_WIN32_SDK_8), join(win32_sdk_folder, 'WindowsKits', '8.1') )
             download_sdk( '%s/%s.tar.gz' % (self.package_path, PACKAGES_WIN32_SDK_10), join(win32_sdk_folder, 'WindowsKits', '10') )
-            download_sdk( '%s/%s.tar.gz' % (self.package_path, PACKAGES_WIN32_TOOLCHAIN), join(win32_sdk_folder, 'MicrosoftVisualStudio14.0') )
+            download_sdk( '%s/%s.tar.gz' % (self.package_path, PACKAGES_WIN32_TOOLCHAIN), join(win32_sdk_folder, 'MicrosoftVisualStudio14.0'), strip_components=0 )
```
`strip_components = 0` to avoid removing the first part of the hierarchy.

JCash committed:
Win32 fixes to the sdk package script

eangelin commented on build_tools/waf_dynamo.py:
```diff
@@ -1284,122 +1286,45 @@ def create_clang_wrapper(conf, exe):
+        conf.fatal("Unable to determine msvc version: '%s'" % msvc_version)
+
+    msvc_path = (os.path.join(msvcdir,'VC', 'Tools', 'MSVC', msvc_version, 'bin', 'Host'+arch, arch),
+                os.path.join(windowskitsdir, '8.1', 'bin', arch))
```
Shouldn't all of these 8.1's be 10's?

eangelin commented on scripts/build.py:
```diff
@@ -65,9 +65,11 @@
+WINDOWS_SDK_10_VERSION="10.0.18362.0"
+WINDOWS_MSVC_2019_VERSION="14.25.28610"
+PACKAGES_WIN32_TOOLCHAIN="Microsoft-Visual-Studio-2019-{0}".format(WINDOWS_MSVC_2019_VERSION)
 PACKAGES_WIN32_SDK_8="WindowsKits-8.1"
```
Shouldn't we remove 8.1 entirely?

eangelin commented on scripts/build.py:
```diff
@@ -503,7 +505,7 @@ def download_sdk(url, targetfolder, strip_components=1):
             win32_sdk_folder = join(self.ext, 'SDKs', 'Win32')
             download_sdk( '%s/%s.tar.gz' % (self.package_path, PACKAGES_WIN32_SDK_8), join(win32_sdk_folder, 'WindowsKits', '8.1') )
```
... and this

eangelin commented on scripts/package/package_win32_sdk.sh:
```diff
@@ -60,12 +60,14 @@ if [ ! -e "${TARGET_PATH}/${PACKAGES_WIN32_TOOLCHAIN}" ]; then
 	TMP=MicrosoftVisualStudio2019
 
```
Not related to this PR, but VC_PATH above assumes Community edition, we should mention in a readme you need to change this if you have Professional. Or even better detect it :)

eangelin commented on scripts/package/package_win32_sdk.sh:
```diff
@@ -60,12 +60,14 @@ if [ ! -e "${TARGET_PATH}/${PACKAGES_WIN32_TOOLCHAIN}" ]; then
+	cp -r -v "$VC_PATH/VC/Tools/MSVC/$MSVC_VERSION/bin/Hostx64/x86" "$TMP/VC/Tools/MSVC/$MSVC_VERSION/bin/Hostx86"
 	cp -r -v "$VC_PATH/VC/Tools/MSVC/$MSVC_VERSION/include" "$TMP/VC/Tools/MSVC/$MSVC_VERSION"
 	cp -r -v "$VC_PATH/VC/Tools/MSVC/$MSVC_VERSION/lib/x64" "$TMP/VC/Tools/MSVC/$MSVC_VERSION/lib"
 	cp -r -v "$VC_PATH/VC/Tools/MSVC/$MSVC_VERSION/lib/x86" "$TMP/VC/Tools/MSVC/$MSVC_VERSION/lib"
```
Also noticed this script doesn't remove the TMP dir 'MicrosoftVisualStudio2019' after tar'ing

JCash committed:
Merge branch 'dev' into issue-4822-win32-package-update

JCash committed:
Removed need for WIndows Kit 8.1

Added sign tool to WIndows Kit 10 package
Started using /local_sdks as the preferred target folder for these packages

JCash committed:
All packages now output to defold/local_sdks

JCash committed:
Updated to Windows Kit 10 for the dmsdk as well

JCash committed:
Merge branch 'issue-4822-win32-package-update' of github.com:defold/defold into issue-4822-win32-package-update
